src/server/middleware/systraceMiddleware.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2017-present, Callstack.
+ * All rights reserved.
+ * 
+ * @flow
+ */
+import type { Middleware, $Request } from 'express';
+
+const fs = require('fs');
+const dedent = require('dedent');
+const logger = require('../../logger');
+
+const systraceMiddleware: Middleware = (req: $Request, res, next) => {
+  if (req.path !== '/systrace') {
+    next();
+    return;
+  }
+
+  const path = `/tmp/react_native_${Date.now()}.json`;
+
+  // $FlowFixMe: rawBodyMiddleware adds `rawBody` to all requests
+  fs.writeFileSync(path, req.rawBody);
```

Comment:
Does it need to be sync? I think using async version will be pretty easy here.

We also need error handling.